examples/microwave-oven-app/microwave-oven-common/include/microwave-oven-device.h

```diff
@@ -70,27 +78,58 @@ class ExampleMicrowaveOvenDevice : public MicrowaveOvenControl::Delegate,
      * handle command for microwave oven control: set cooking parameters
      */
     Protocols::InteractionModel::Status HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime,
-                                                                           uint8_t powerSetting) override;
+                                                                           bool startAfterSetting, Optional<uint8_t> powerSetting,
+                                                                           Optional<uint8_t> wattSettingIndex) override;
 
     /**
      * handle command for microwave oven control: add more time
      */
     Protocols::InteractionModel::Status HandleModifyCookTimeCallback(uint32_t finalCookTime) override;
 
+    /**
+     *   Get the watt setting from the supported watts array.
+     *   @param index The index of the watt setting to be returned. It is assumed that watt setting are indexable from 0 and with no
+     * gaps.
+     *   @param wattSetting A reference to receive the watt setting on success.
+     *   @return Returns a CHIP_NO_ERROR if there was no error and the label was returned successfully.
+     *   CHIP_ERROR_NOT_FOUND if the index in beyond the list of available labels.
+     */
+    CHIP_ERROR GetWattSettingByIndex(uint8_t index, uint16_t & wattSetting) override;
+
+    /**
+     * Get the value of PowerSetting.
+     */
+    uint8_t GetPowerSetting() const override { return mPowerSetting; }
+
     /**
      * Get the value of MinPower.
      */
-    uint8_t GetMinPower() const override { return kDefaultMinPower; }
+    uint8_t GetMinPower() const override { return kMinPower; }
 
     /**
      * Get the value of MaxPower.
      */
-    uint8_t GetMaxPower() const override { return kDefaultMaxPower; }
+    uint8_t GetMaxPower() const override { return kMaxPower; }
 
     /**
      * Get the value of PowerStep.
      */
-    uint8_t GetPowerStep() const override { return kDefaultPowerStep; }
+    uint8_t GetPowerStep() const override { return kPowerStep; }
+
+    /**
+     * Get the value of MaxCookTime.
+     */
+    uint32_t GetMaxCookTime() const override { return kMaxCookTime; }
```

Comment:
please make this return a Duration (that includes unit of measurement) or make the method have a `Sec/Ms/Min/Hr/...` suffix

Reply:
This should have been done throughout this class - I would feel uneasy about asking it as part of this PR however if not done, please at least create an issue. We should have names like `cookTimeSec` instead of `cookTime` throughout and any method saying cookTime should be cookTimeSec as well.

Reply:
Already changed all the variables and methods name with adding unit for cookTimeSec. But seems no need to create an issue... may I ask is it just create for changing all the 'cookTime' name to 'cookTimeSec'? 

> please at least create an issue